src/shared/splice-array.js

```diff
@@ -0,0 +1,20 @@
+const myPush = require('./push');
+
+function mySplicedArray(array, start, count) {
+  const { length } = array;
+  const newArray = [];
+  let newCount = count;
+  if (newCount > 0) {
+    newCount -= 1;
+  } else {
+    return array;
+  }
+  for (let i = 0; i < length; i += 1) {
+    if (!((i <= start + newCount && i >= start) || (i <= start && i >= start + newCount))) {
```

Comment:
Would better to split something like that.
Hard to read and not really clean. (Does its job though,,, so goood job! )